xchange-binance/src/main/java/info/bitrich/xchangestream/binance/BinanceStreamingMarketDataService.java

```diff
@@ -72,8 +71,10 @@
     private final AtomicBoolean fallenBack = new AtomicBoolean();
     private final AtomicReference<Runnable> fallbackOnApiCall = new AtomicReference<>(() -> {});
 
-    public BinanceStreamingMarketDataService(BinanceStreamingService service, BinanceMarketDataService marketDataService, Runnable onApiCall) {
+    public BinanceStreamingMarketDataService(BinanceStreamingService service, BinanceMarketDataService marketDataService, Runnable onApiCall,
```

Comment:
For consistency, please follow the pattern used by `onApiCall`: extract the setting as an integer in `BinanceStreamingExchange.initServices` and then pass that integer here.

If you also perform the validation in `initServices` it will really simplify the code.  You can simply default the value there to 1000ms (use an `int`, not an `Integer`, so it can't be null) which means you don't need a lot of complex nested conditions when building the subscription string.